pkg/extension/sumologicextension/extension.go

```diff
@@ -986,10 +986,13 @@ func addClientCredentials(req *http.Request, credentials accessCredentials) {
 	req.Header.Add("Authorization", authHeaderValue)
 }
 
+// TODO(ck): hostname allows the darwin tests to bypass fqdn.
+var hostname = fqdn.FqdnHostname
+
 // getHostname returns the host name consistently with the resource detection processor's defaults
 // TODO: try to dynamically extract this from the resource processor in the pipeline
```

Comment:
is it possible to do what this comment says? i.e extract the hostname from the resource processor.

Reply:
Good question. I'd be surprised if extensions had a way to get processor state like that. I also doubt that'd be a productive way to go at least as far as tests are concerned. (i.e. setting up an entire pipeline with the resource detection processor configured to not use os instead of dns.)

Side note, I wager that processor has the same issue (when configured with dns)

Reply:
I guess that solution would assume that we'd be using the resource processor in the pipeline like below.

```
service:
  extensions:
    - sumologic
  pipelines:
    logs:
      receivers: 
        - tcplog
      processors:
        - resource
```

Reply:
@rnishtala-sumo  you okay with kicking this can down the road? Maybe noting that option in https://github.com/SumoLogic/sumologic-otel-collector/issues/1295

Reply:
yes thats fine